### ISSUE TITLE
some issues about the real robot test

### ISSUE THREAD

**Issue by ShaohangXu** [yxyang/locomotion_simulation#5]
Hi Yang @yxyang. Thanks for your code sharing.

I tested the code on the real robot with TX2. It worked well in sim but I did not get good results on the real robot.

I wrote a stand controller which sends initial motor angles. Besides, there is a one-second trot gait controlled by your mpc-based controller. I attached two tested videos in [Baidu Yun](https://pan.baidu.com/s/1jJLdfyQu7gR21jKiufEOMQ). The password is 0h7a

One problem is that the first control step will finish after ~0.2s when the robot reset. This causes that the robot is in protection mode, which will brake the motors. I have to pull it up myself in case the robot falls down. Sometimes other control step would exteed the safe frequency (33Hz) too. Have you met this problem?

Another problem is that the robot stamped too heavily and almost fell when troting. I have to pull it up to protect it. The control frequency is ~150Hz, which is less than 300Hz you recommended. Will the frequency affect this so much? If it runs in ~300Hz, how steady it will be?

The official version of A1 has only TX2. May I ask how you change it to Agx Xavier? Is there any problem when installing it?

You said if turned to C++, the code would run faster. May I ask how to change it? Will it be possible using ROS? Or just pure C++ projects using the Unitree C++ SDK？

By the way, may I ask whether you have trained any RL-based controller through the a1 gym env?

**Comment by yxyang** [comment]
1) For torque control, a control frequency of 300Hz is probably needed, and
below it would be very unstable.
2) I simply mounted the Xavier on top of the robot. Not the most integrated
setup, but seemed to work fine. Also I found a better solution to use
Apple's Mac Mini with M1, which is ~2x faster compared to Xavier and
consumes less power.
3) C++: ROS is not needed though it may provide additional benefits. I'm
not a C++ expert so can't give much advice on this...
4) Yes, see the end of this video
<https://www.youtube.com/watch?v=agO22KiWiBc&ab_channel=YuxiangYang> for an
example.

Feel free to add my WeChat: (DavisYang_nfls).

Best,
Yuxiang
